### ISSUE TITLE
Redundant casts in string CONCAT

### ISSUE THREAD

**Issue by vegarsti** [duneanalytics/harmonizer#91]
After SQLGlot added SafeConcat (https://github.com/tobymao/sqlglot/pull/1738), and we bumped Harmonizer to use that version, we started seeing very verbose `CONCAT` calls.

We should set `STRICT_STRING_CONCAT = FALSE` for DuneSQL!

**Comment by vegarsti** [comment]
This doesn't look like it's actually the issue, with this it gets even worse! I'd like to get rid of both COALESCE and CAST calls here:

Original
```
select CONCAT('\x', 'lol')
```

With `STRICT_STRING_CONCAT = False` it's

```
SELECT
  CONCAT(
    CAST(COALESCE(COALESCE('x', ''), '') AS VARCHAR),
    CAST(COALESCE(COALESCE('lol', ''), '') AS VARCHAR)
  )
```
and with `False` it's 
```
SELECT
  CONCAT(
    CAST(CAST(COALESCE(COALESCE('x', ''), '') AS VARCHAR) AS VARCHAR),
    CAST(CAST(COALESCE(COALESCE('lol', ''), '') AS VARCHAR) AS VARCHAR)
  )
```

**Comment by georgesittas** [comment]
@vegarsti we decided to refactor this, so this issue should be resolved as well when we get 2639 in.

EDIT: now available in v20.1.0

**Comment by vegarsti** [comment]
Nice, well done! We're not in need of harmonizer anymore, but thanks for letting us know!
